### PR TITLE
(#563) Add documentation about hooks

### LINKED ISSUE #563: Add documentation for hook scripts.
## What New Or Updated Would You Like To See?

A documentation page about hook scripts.

## Why Is It Needed?

Because they are not currently documented.

## Additional Context?

Hook feature added into Chocolatey CLI in chocolatey/choco#2535, expected to go out in v1.2.0

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
## Description Of Changes

Adds documentation about hooks

## Motivation and Context

See #563

Should not be merged until hooks go out in a Chocolatey CLI release.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [ ] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [x] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left hand side)/
* [ ] Menu structure has been updated

## Related Issue

Fixes #563

### CODE REVIEW

File: input/en-us/features/hook.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+xref: hooks
+Title: Extend Chocolatey With PowerShell Scripts (Hooks)
+Description: Run PowerShell before and after Package Automation Scripts
+---
+
+A feature is available in Chocolatey CLI v1.2.0, which allows you to extend a package's automation script(s) with additional PowerShell scripts, called hooks. 
+
+## Overview
+
+Hooks allow you to run code before a package automation script like a `chocolateyInstall.ps1` runs, or after. They even work when automation scripts are not included in the package. Each script can selectively target a single package ID, or run for all packages. Then distribute them using `.hook` packages.
```
Comment (corbob) [inline input/en-us/features/hook.md:12]: This last sentence feels a little out of place.

```suggestion
Hooks allow you to run code before a package automation script like a `chocolateyInstall.ps1` runs, or after. They even work when automation scripts are not included in the package. Each script can selectively target a single package ID, or run for all packages. Then they can be distributed using `.hook` packages.
```

File: input/en-us/features/hook.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Creating Hook Scripts
+
+Hook script are PowerShell `.ps1` files, and are run in the same environment as package automation scripts are, so they can access all the same environment variables and helpers.
```
Comment (corbob) [inline input/en-us/features/hook.md:16]:
```suggestion
Hook scripts are PowerShell `.ps1` files, and are run in the same environment as package automation scripts run, so they have access to all of the same environment variables and helpers. 
```

File: input/en-us/features/hook.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Hooks allow you to run code before a package automation script like a `chocolateyInstall.ps1` runs, or after. They even work when automation scripts are not included in the package. Each script can selectively target a single package ID, or run for all packages. Then distribute them using `.hook` packages.
+
+## Creating Hook Scripts
+
+Hook script are PowerShell `.ps1` files, and are run in the same environment as package automation scripts are, so they can access all the same environment variables and helpers. 
+
+The conditions for when a hook is run is based on the filename of the hook. `<pre|post>-<install|beforemodify|uninstall>-<packageID|all>.ps1` is the filename format. The first section is for timing of when the hook script runs. Filenames that start with `pre` run before the package automation script (e.g. the `chocolateyInstall.ps1`), while `post` scripts are run afterwards. The second section of the filename determines which package automation script the hook is run before or after. The third section of the filename is either a string of the package ID for which the hook should be run, or the keyword `all` to specify that the hook should be run for all package IDs.
```
Comment (corbob) [inline input/en-us/features/hook.md:18]:
```suggestion
The conditions for when a hook is run is based on the filename of the hook. The filename format is `<pre|post>-<install|beforemodify|uninstall>-<packageID|all>.ps1`. The first section is for the timing of when the hook script runs. Filenames that start with `pre` run before the package automation script (e.g. the `chocolateyInstall.ps1`), while filenames that start with `post` are run afterwards. The second section of the filename determines which package automation script the hook is run before or after. The third section of the filename is either a string of the package ID for which the hook should be run, or the keyword `all` to specify that the hook should be run for all package IDs.
```

File: input/en-us/features/hook.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+1. Create a package with a `.hook` suffix like `choco new name.hook`
+1. Delete the created files and folders except for the `.nuspec`
+1. Create a `hook` folder in the root of the package (next to the `.nuspec`).
+1. Put your hook PowerShell script files inside the `hook` folder.
+1. Pack and install your new Hook package.
+
+
+## Recommenations
+
+## Still under development
```
Comment (corbob) [inline input/en-us/features/hook.md:36]: This is supposed to be a subheading of Recommendations right?

```suggestion
### Still under development
```

File: input/en-us/features/hook.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+Hook scripts and packages are still new, so guidelines and best practices are still being determined. Follow along at the [Chocolatey Community Hooks Repository](https://github.com/chocolatey-community/chocolatey-hooks) for the latest information.
+
+## Avoid taking dependencies on regular packages
```
Comment (corbob) [inline input/en-us/features/hook.md:40]: This is supposed to be a subheading of Recommendations right?

```suggestion
### Avoid taking dependencies on regular packages
```

File: input/en-us/features/hook.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+It is best to avoid having a `.hook` package taking a dependency on regular packages. This is so the included hooks from the package can be installed before other packages get initially installed. For example, if there is a `pre-install-git.ps1` hook and the `.hook` package it is included in takes a dependency on `git`, then the `chocolateyInstall.ps1` from the `git` package would have already run before the hook is installed, which is counterproductive.
+
+## Do take dependencies on extension packages
```
Comment (corbob) [inline input/en-us/features/hook.md:44]: This is supposed to be a subheading of Recommendations right?

```suggestion
### Do take dependencies on extension packages
```

**Review by corbob — CHANGES_REQUESTED** [review]
Just a few minor changes.

File: input/en-us/features/hook.md (hunk 1 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Yup, change looks good.

File: input/en-us/features/hook.md (hunk 4 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Fixed

File: input/en-us/features/hook.md (hunk 6 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Fixed

File: input/en-us/features/hook.md (hunk 5 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Fixed

File: input/en-us/features/hook.md (hunk 3 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Fixed

File: input/en-us/features/hook.md (hunk 2 above)
Comment (TheCakeIsNaOH) [reply input/en-us/features/hook.md]: Fixed

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@TheCakeIsNaOH this looks great! Thanks for getting this new page added, and for the hook feature in general!

**Pull request merged**